packages/shared/src/components/squads/SquadMemberMenu.tsx

```diff
@@ -174,7 +174,9 @@ export default function SquadMemberMenu({
     });
 
     const canRemoveMember =
-      canUpdateRole || verifyPermission(squad, SourcePermissions.MemberRemove);
+      canUpdateRole ||
+      (verifyPermission(squad, SourcePermissions.MemberRemove) &&
+        member.role === SourceMemberRole.Member);
```

Comment:
Just a note, from what I can see this is covered on the API with `requireGreaterAccessPrivilege` which kinda spoils our clean usage of `verifyPermission`. As soon as we are actually checking roles and not permissions it could lead to API vs UI permission inconsistencies. 

I think right think here would be to have two separate permissions for our current implementation but I do know we are deep in v4 so might wanna leave it for later.

Reply:
Hmm yeah it's difficult as that would need to evaluate PER member and become very expensive to do API side as well.
Think for now this would be fine as it's the exact requirement. we can always optimize down the line.

Also tagging @sshanzel for his opinion as I'm open to adding a fn on FE side as well.

Reply:
The only thing I can think of is the `roleRank`. If the user has a greater numerical value, we show the option.

Reply:
The specs say you can only do members though, so not sure that would be valid.

Reply:
But as a moderator, you have greater access over members only, right? So the condition won't have equal. For owners though, this probably should always be shown.

Reply:
Ah yes, but then it's less explicit when we add new roles right?
Thus this being the more safe evaluation for now.

But still happy to promote to a function if that deems needed.

Reply:
I think core issue is from the API side so for frontend implementation I would leave it as it is.

Reply:
If we introduce a new role and if you have greater access privileges, you should have power over it, right? Though would probably depend on the product's decision. So having it explicit now should work with the requirements. We can update if necessary.

Reply:
Depends on product indeed, for now it's not set but could cause to break if product decided not to have that option :)